chore: release April-May newsletter (#1135)
Financial-Times/origami: apps/website/_posts/2023-05-31-newsletter.md

---
title: Origami Newsletter, April 2023 - May 2023
description: Origami is preparing two talks in upcoming month. We released FT Professional themes for o-buttons and o-forms. We started working on Origami For Everyone(O3) and we started talking to other organisations to share knowledge about design systems.
author: Akaki Mikaia
tags:
  - Newsletter
---

<abbr title="Too long; didn't read">
<strong>
TL;DR:
</strong>
</abbr> {{page.description}}

## Top things

Some of the bigger Origami news since our last update:

## Upcoming talks and workshops

Mark your calendars! We have two engaging talks lined up for the upcoming month:

1. Introduction to Origami: Lee will be delivering an insightful session that provides a high-level overview of Origami. Whether you work in Product or Technology, this talk is for you! Discover why Origami exists and how it can benefit you in your work.

> 2. Origami Foundations: Join Ako as he explores the fundamentals of Origami components. This talk will delve into the usage, advantages, and practical implementation of key components such as o-normalise, o-fonts, o-spacing, o-colors, o-typography, and a glimpse into o-grid.

Author: [1c70770](https://github.com/Financial-Times/origami/pull/1135/commits/1c70770087c3d483e6da85bef77dd87d9e485659) Just added this tiny sentence? do you think this is better?

> 2. Origami Foundations: Join Ako as he explores the fundamentals of Origami components. This talk will delve into the usage, advantages, and practical implementation of key components such as o-normalise, o-fonts, o-spacing, o-colors, o-typography, and a glimpse into o-grid.

Reviewer: That should go under (2) for your talk, as mine isn't engineer focused, but otherwise yes 👍

> 2. Origami Foundations: Join Ako as he explores the fundamentals of Origami components. This talk will delve into the usage, advantages, and practical implementation of key components such as o-normalise, o-fonts, o-spacing, o-colors, o-typography, and a glimpse into o-grid.

Author: done [526e8e0](https://github.com/Financial-Times/origami/pull/1135/commits/526e8e09103660f7d7b240f3dcddbecd77863634)

## FT Professional

Last months Origami has been working with FT Professional to gather requirements and identify which Origami components require a new professional theme, this quarter we managed to release more themes than we promised and we will release few more before Q2 finishes.

Currently we support `o-banner`, `o-forms`, `o-tooltip`, `o-typography` and `o-buttons` component themes and variations.

## Origami for everyone

> We're excited about Origami for Everyone, a project that aims to address these challenges and make the Origami design system more integral to the delivery of every user-facing product that FT delivers. The project has three main goals: universal usability, designer-developer collaboration, and multi-platform compatibility.

Reviewer: I wasn't so sure on this section as it assumes some knowledge, "address these challenges" doesn't mention what challenges. I'm also thinking the goals cover more the what more than the why, which might be good to restate here. Feel free to throw away my suggestion but pillaging from different parts of Rhy's announcement, how about something like this?

```suggestion
We're excited about [Origami for Everyone](https://financialtimes.atlassian.net/wiki/spaces/OR/blog/2023/05/18/8064925807/Introducing+Origami+for+Everyone) as creating differentiated products targeting different audiences is a central theme of the [FT’s product and growth strategy](https://inside.ft.com/home/knowledge-base/Product-strategy-2023). Origami for Everyone will allow Origami to scale to support more brands, designer-developer collaboration, and multi-platform compatibility.
```

> To achieve these vision, we'll need to make some fundamental changes in the existing Origami implementation. We've already started recruiting a new designer to lead on the design side of Origami, and we've been conducting research into design tokens, which will form the backbone of the new implementation of Origami. Ben already did a demo for new design tokens system and we got a positive feedback from our potential users. You can read more about this on our blogpost [Origami Team Trials Design Tokens](https://financialtimes.atlassian.net/wiki/spaces/OR/blog/2023/04/20/8047034476/Origami+Team+Trials+Design+Tokens) and if you want to know more of a technical decisions made along the way check out [Conventions for origami Design Tokens](https://financialtimes.atlassian.net/wiki/spaces/OR/blog/2023/04/26/8049524875/Conventions+for+Origami+Design+Tokens).

Reviewer: ```suggestion
To achieve these vision, we'll need to make some fundamental changes in the existing Origami implementation. We've already started recruiting a new designer to lead on the design side of Origami, and we've been conducting research into design tokens, which will form the backbone of the new implementation of Origami. Ben already did a demo for new design tokens system and we got some initial positive feedback from our potential users. You can read more about this on our blogpost [Origami Team Trials Design Tokens](https://financialtimes.atlassian.net/wiki/spaces/OR/blog/2023/04/20/8047034476/Origami+Team+Trials+Design+Tokens) and if you want to know more of a technical decisions made along the way check out [Conventions for origami Design Tokens](https://financialtimes.atlassian.net/wiki/spaces/OR/blog/2023/04/26/8049524875/Conventions+for+Origami+Design+Tokens).
```

> We encourage everyone to actively participate in this project, offering feedback and sharing proposals for the new Origami architecture. Together, we can shape a future where the Origami design system works seamlessly with every product we deliver.

Reviewer: Strong. I like it.

## Knowledge sharing with other organisations

> We started talking to other organisations to share knowledge about design systems. We had a meeting with the design system team from the Lego and Washington Post. Next month we have a meeting scheduled with Conde Nast and we are all excited to share our knowledge and learn from each other.

Reviewer: Little tweak here as it implies we haven't had similar conversations in the past and also to include Compare the Market!

```suggestion
We've been talking with other organisations to share knowledge about design systems. We had a meeting with the design system team from the Lego, Washington Post, and Compare The Market. Next month we have a meeting scheduled with Conde Nast and we are all excited to share our knowledge and learn from each other.
```

## Special Thanks

> We had few contributors who helped us to improve Origami components and we want to thank them for their contributions. Thanks to Juan Sanchez Alcala for contributing to `o-topper`, `o-video` and `o-comments` components.

Reviewer: Without "a" this reads as "not many" 😅 

```suggestion
We had a few contributors who helped us to improve Origami components and we want to thank them for their contributions. Thanks to Juan Sanchez Alcala for contributing to `o-topper`, `o-video` and `o-comments` components.
```

Special thanks to Anna Bebb for deprecating  the `n-sliding-popup` component – that prompted us to write some docs on how to do deprecate Origami components. Now we have a [guide on how to deprecate Origami components](https://origami.ft.com/documentation/components/versioning/#deprecate-a-component)

We had new contributors who helped us to improve Origami components. Thanks to Marlon Bucley for helping us to improve `o-multi-select` component and thanks to Ed Gargan for helping us to improve `o-forms` component.

> 

Reviewer: ```suggestion

And a final thankyou to Arjun Gadhia who contributed a new customisation feature to o-autocomplete to support the TagMe rewrite. This was a most [excellent pull request](https://github.com/Financial-Times/origami/pull/1127) which included problem context, screenshots, documentation, and demos – making it very easy for the team to review and approve.
```

## Broader update

A digest list of some other things that have happened in April and May:

> - Major [o-multi-select](https://registry.origami.ft.com/components/o-multi-select): After some feedback from Spark we improved `o-multi-select` functionality and now component can be initialised with a selected values. We also added custom event `OptionChange` to enable users to track interactions to multi-select component. We also published TSX templates for multi select.

Reviewer: ```suggestion
- Major [o-multi-select](https://registry.origami.ft.com/components/o-multi-select): After some feedback from Spark we improved `o-multi-select` functionality and now it can be initialised with a selected value. We also added a custom event `OptionChange` to enable users to track interactions to o-multi-select. We also published TSX templates.
```

> - Minor [o-forms](https://registry.origami.ft.com/components/o-forms): `o-forms` had few minor releases in the last two months. We removed "green" valid state from `o-forms`. Deprecated pseudo-radio-links. `o-forms` now also has TSX templates published and can be used by other TSX projects. We also released FT Professional themes for `o-forms` elements. `o-forms` does not error any more when no themes given for primary mixin. Now we apply invalid input styles to inputs without browser validation.

Reviewer: ```suggestion
- Minor [o-forms](https://registry.origami.ft.com/components/o-forms): `o-forms` had a few minor releases in the last two months. We removed the "green" valid state from `o-forms`. Deprecated pseudo-radio-links. `o-forms` now also has TSX templates published and can be used by other TSX projects. We also released FT Professional themes for `o-forms` elements. Now we apply invalid input styles to inputs with custom validation, as well as those which use default browser validation.
```

> - Minor [o-topper](https://registry.origami.ft.com/components/o-topper): We changed outline color for focus links under matisse theme.

Reviewer: Good to highlight why and to use British English in writing outside of code. 
https://origami.ft.com/documentation/principles/tone-and-language/#use-british-english-for-documentation

```suggestion
- Minor [o-topper](https://registry.origami.ft.com/components/o-topper): To improve accessibility, we changed the outline colour for focus links in some cases.
```

- Minor [ft-concept-button](https://registry.origami.ft.com/components/ft-concept-button): Now `ft-concept-button` sets aria-label on initialise.
- Minor [o-buttons](https://registry.origami.ft.com/components/o-buttons): Now o-buttons supports `professional` themes.

> - Minor [o-colors](https://registry.origami.ft.com/components/o-colors): Professional color Mint is not part of o-colors pallet. We added `page-inverse` and `base-inverse` colour usecases.

Reviewer: ```suggestion
- Minor [o-colors](https://registry.origami.ft.com/components/o-colors): Professional color Mint is now part of o-colors pallet. We added `page-inverse` and `base-inverse` colour usecases.
```

- Minor [o-tooltip](https://registry.origami.ft.com/components/o-tooltip): added `professional` themes and now it supports setting themes imperatively.
- Minor [o-typography](https://registry.origami.ft.com/components/o-typography): added `professional` themes for body and links.
- Minor [o-autocomplete](https://registry.origami.ft.com/components/o-autocomplete): added a `suggestionTemplate` override option. This function is used to override the default rendering of suggestion items in autocomplete, with a function that returns a custom HTML string for the given option. It is typically used when wanting to provide additional context or styling for each suggestion item.
- Minor [o-banner](https://registry.origami.ft.com/components/o-banner): added a new `inverse-professional` themes.

> - Patch [o-header](https://registry.origami.ft.com/components/o-header): We improved scroll buttons behaviors on o-header subnav.

Reviewer: ```suggestion
- Patch [o-header](https://registry.origami.ft.com/components/o-header): We improved scroll button behaviour in the o-header subnav.
```
